pepper-apis/dsm-core/src/main/java/org/broadinstitute/dsm/model/elastic/export/tabular/FilterExportConfig.java

```diff
@@ -2,28 +2,114 @@
+        if (childQuestions != null) {
+            this.childConfigs = IntStream.range(0, childQuestions.size())
+                    .mapToObj(qNum -> new FilterExportConfig(this, childQuestions.get(qNum), qNum)).collect(Collectors.toList());
+        }
+    }
+
+    /** short constructor for getting a config for a subquestion of a composite question */
+    private FilterExportConfig(FilterExportConfig parent, Map<String,Object> childQuestion, int childIndex) {
+        this.parent = parent.getParent();
+        this.type = (String) childQuestion.get(ESObjectConstants.QUESTION_TYPE);
+        String childStableId = (String) childQuestion.get(ESObjectConstants.STABLE_ID);
+        if (StringUtils.isBlank(childStableId)) {
+            // some subquestions don't have stableIds (e.g. RELEASE_MINOR_PHYSICIAN from PanCan), we need to ensure the stableId
+            // is something unique to avoid values getting overwritten
+            childStableId = (String) childQuestion.get(ESObjectConstants.QUESTION_TEXT);
+            if (StringUtils.isNotBlank(childStableId)) {
+                childStableId = childStableId.toUpperCase().replace(" ", "_");
+            } else {
+                childStableId = "SUBQUESTION" + childIndex;
+            }
+        }
+        this.column = new ParticipantColumn(childStableId, parent.column.getTableAlias());
+        String childDisplayText = (String) childQuestion.get(ESObjectConstants.QUESTION_TEXT);
+        if (StringUtils.isBlank(childDisplayText)) {
+            childDisplayText = childStableId;
+        }
+        this.column.setDisplay(childDisplayText);
+
+        this.questionDef = childQuestion;
+        this.options = getOptionsForQuestion(childQuestion);
+    }
+
+    public boolean isAllowMultiple(Map<String, Object> questionDef) {
+        if (questionDef != null && questionDef.get(ESObjectConstants.ALLOW_MULTIPLE) != null) {
+            return (boolean) questionDef.get(ESObjectConstants.ALLOW_MULTIPLE);
+        }
+        return false;
+    }
+
+    private List<Map<String, Object>> getOptionsForQuestion(Map<String, Object> questionDef) {
+        List<Map<String,Object>> options = (List<Map<String, Object>>) questionDef.get(ESObjectConstants.OPTIONS);
+        if (questionDef.containsKey(ESObjectConstants.OPTION_GROUPS)) {
+            Object groups = questionDef.get(ESObjectConstants.OPTION_GROUPS);
+            if (groups instanceof List) {
+                for (Map<String, Object> group : (List<Map<String, Object>>) groups) {
+                    if (group.containsKey(ESObjectConstants.OPTIONS) ) {
+                        options.addAll((List<Map<String, Object>>) group.get(ESObjectConstants.OPTIONS));
+                    }
+                }
+            }
+        }
+        return options;
+    }
+
+    public List<Map<String, Object>> getChildQuestions() {
+        return questionDef != null ? (List<Map<String, Object>>) questionDef.get(ESObjectConstants.CHILD_QUESTIONS) : null;
```

Comment:
I think it is better technique to return empty list rather than `null` from the method which has to return list, it will help to avoid ugly if check in the place where this method is called.

Reply:
I can see that -- for now I'm mimicking the structure of the data returned from ElasticSearch, so leaving `null` where there are no childQuestions